coresdk/src/backend/text_driver.cpp

```diff
@@ -131,7 +171,12 @@ void sk_draw_text(
         _sk_draw_bitmap_text(surface, x, y, text, clr);
         return;
     }
-    if (! (font && font->_data)) return; // error with font
+    if (! (VALID_PTR(font, FONT_PTR))) return; // error with font
+
+    // Check that the font size has been loaded and create if not
+    if (font->_data.count(font_size) == 0) {
+        font->_data[font_size] = TTF_OpenFont(font->filename.c_str(), font_size);
```

Comment:
This one needs to stay I think, because the helper returns if the font for a size already exists, but we want the student to be able to call draw text with a valid font object but without having had loaded that font size?